components/Dialog/Dialog.unit.test.jsx

```diff
@@ -54,7 +47,7 @@ describe('Dialog', () => {
     });
 
     const onCloseMock = jest.fn();
-    mount(<Dialog onClose={onCloseMock} />);
+    render(<Dialog onClose={onCloseMock} />);
```

Comment:
Here is missing a test when another key is pressed. My suggestions is something like:

`
eventMap.keydown({ key: 'Enter' });
eventMap.keydown({ key: 'Escape' });

expect(onCloseMock).toHaveBeenCalledTimes(1);
`

or another test to check if function is called when another key is pressed.

Reply:
About the dropdown, didn't manage to cover all this scenarios, but covered a few more.